### PR TITLE
checking the system for dynamics and energy presence

### PR DESCRIPTION
**Pull request by kzqureshi**
Check for the system's energy and system's dynamic is added.

### CODE REVIEW

**Comment by kzqureshi** [comment]
- [ ] Test for py checks

**Comment by kzqureshi** [comment]
Changes from ubermag/micromagnetictests#47 required for proper test for the relaxdriver.

File: mumax3c/drivers/mindriver.py [hunk 1]
```diff
@@ -49,6 +49,11 @@ class MinDriver(Driver):
     def _checkargs(self, **kwargs):
         pass  # no kwargs should be checked
 
+    def _check_system(self, system):
+        """Checks the system has energy in it"""
+        if len(system.energy) == 0:
+            raise AttributeError("System's energy is not defined")
```
Comment (samjrholt) [inline mumax3c/drivers/mindriver.py:55]:
```suggestion
            raise RuntimeError("System's energy is not defined")
```

File: mumax3c/drivers/relaxdriver.py [hunk 2]
```diff
@@ -49,6 +49,13 @@ class RelaxDriver(Driver):
     def _checkargs(self, **kwargs):
         pass  # no kwargs should be checked
 
+    def _check_system(self, system):
+        """Checks the system has dynamics in it"""
+        if len(system.dynamics) == 0:
+            raise AttributeError("System's dynamics is not defined")
```
Comment (samjrholt) [inline mumax3c/drivers/relaxdriver.py:55]:
```suggestion
            raise RuntimeError("System's dynamics is not defined")
```

File: mumax3c/drivers/relaxdriver.py [hunk 3]
```diff
@@ -49,6 +49,13 @@ class RelaxDriver(Driver):
...
+        if len(system.energy) == 0:
+            raise AttributeError("System's energy is not defined")
```
Comment (samjrholt) [inline mumax3c/drivers/relaxdriver.py:57]:
```suggestion
            raise RuntimeError("System's energy is not defined")
```

File: mumax3c/drivers/timedriver.py [hunk 4]
```diff
@@ -62,6 +62,13 @@ def _checkargs(self, **kwargs):
             msg = f"Cannot drive with {n=}."
             raise ValueError(msg)
 
+    def _check_system(self, system):
+        """Checks the system has dynamics in it"""
+        if len(system.dynamics) == 0:
+            raise AttributeError("System's dynamics is not defined")
```
Comment (samjrholt) [inline mumax3c/drivers/timedriver.py:68]:
```suggestion
            raise RuntimeError("System's dynamics is not defined")
```

File: mumax3c/drivers/timedriver.py [hunk 5]
```diff
@@ -62,6 +62,13 @@ def _checkargs(self, **kwargs):
...
+        if len(system.energy) == 0:
+            raise AttributeError("System's energy is not defined")
```
Comment (samjrholt) [inline mumax3c/drivers/timedriver.py:70]:
```suggestion
            raise RuntimeError("System's energy is not defined")
```

**Review by samjrholt — CHANGES_REQUESTED** [review]

File: mumax3c/drivers/relaxdriver.py [hunk 6]
```diff
@@ -49,6 +49,13 @@ class RelaxDriver(Driver):
     def _checkargs(self, **kwargs):
         pass  # no kwargs should be checked
 
+    def _check_system(self, system):
+        """Checks the system has dynamics in it"""
+        if len(system.dynamics) == 0:
+            raise RuntimeError("System's dynamics is not defined")
```
Comment (lang-m) [inline mumax3c/drivers/relaxdriver.py:55]: Relax does not need a dynamics equation

```suggestion
```

**Review by samjrholt — APPROVED** [review]

**Pull request merged**